src/Chainweb/BlockHeader.hs

```diff
@@ -276,53 +279,106 @@ powTarget
+
+    -- A special case for starting a new devnet. Using maxtarget results in an
+    -- two high block production and consecutively orphans and network
+    -- congestion. The consequence are osciallations to take serval hundred
+    -- blocks before the system stabilizes. This code cools down initial block
+    -- production.
+    | ver == Development && _blockHeight p == 1 = EpochStartTime (_bct $ _blockCreationTime p)
+
+    -- New Graph: the block time of the genesis block isn't accurate, we thus
+    -- use the block time of the first block on the chain. Depending on where
+    -- this is within an epoch, this can cause a shorter epoch, which could
+    -- cause a larger difficulty and a reduced target. That is fine, since new
+    -- chains are expected to start with a low difficulty.
+    | parentIsFirstOnNewChain = EpochStartTime (_bct $ _blockCreationTime p)
+
+    -- End of epoch, DA adjustment (legacy version)
     | isLastInEpoch p && oldTargetGuard ver (_blockHeight p) = EpochStartTime bt
+
+    -- End of epoch, DA adjustment
     | isLastInEpoch p = EpochStartTime (_bct $ _blockCreationTime p)
-    | ver == Development && _blockHeight p == 1 = EpochStartTime (_bct $ _blockCreationTime p)
-        -- this is a special case for starting a new devnet. The creation time
-        -- of the development genesis blocks way in the past which would cause
-        -- the first /and/ second epochs to be at max target. By using the first
-        -- block instead of the genesis block for computing the first epoch time
-        -- we slow down mining eariler.
-    | otherwise = _blockEpochStart p
+
+    -- Within epoch
+    | fixedEpochStartGuard ver (_blockHeight p + 1) = _blockEpochStart p
+    | otherwise = _blockEpochStart p .+^ timeBlocked
   where
     ver = _chainwebVersion p
+    cid = _chainId p
+
+    -- 1. timeBlock < _blockCreationTime h .-. _blockCreationTime p
+    --    and thus, epochStart < _blockCreationTime p
+    --
+    -- 2. this can only increase difficulty but not reduce difficulty compared
+    --    to the slowest chain.
+    --
+    -- 3. Difficulty adjustment for the slowest chain is just like difficulty
+    --    adjustment without timeBlocked, assuming that the slowest chain is
+    --    always the slowest chain.
+    --
+    -- 4. Assuming that not a single chain is always the slowest difficulty
+    --    adjustement is will adjust for somewhat higher difficulty than
+    --    targeted and thus for a blockrate that is slower than the target.
+    --
+    --    How can we adjust for that?
+    --
+    --    a) using a constant factor that is based on properties of the
+    --       geometric distribution.
+    --    b) adjusting timeBlocked in both directions, possibly based off the
+    --       avg?
+    --
+    timeBlocked = maximum adjCreationTimes .-. _blockCreationTime p
+        -- the maximum is at least @_blockCreationTime p@ and thus the result is
```

Comment:
Shouldn't this comment go above `timeBlocked` with the previous comment?

Reply:
In the code base we use the convention that comments that are indented refer to the code above. I put this comment separate, because the comment above is about the overall algorithm, while this comment is an implementation detail.

In any case, this code, including the comment will change in the next iteration of the 20-chain work.